presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarballController.java

```diff
@@ -0,0 +1,138 @@
+    {
+        File tarTempFile = getPackageFile(packageUrl);
+        try {
+            tarInstall(tarTempFile);
+            addConfigFiles(CONFIG_DIR);
+            addConnectors(CATALOG_DIR);
+        }
+        finally {
+            if (!tarTempFile.delete()) {
+                LOGGER.warn("Failed to delete the temporary file: %s", tarTempFile.toString());
+            }
+        }
+        LOGGER.debug("Successfully installed Presto");
+    }
+
+    private void tarInstall(File tarFile)
+            throws PrestoManagerException
+    {
+        try {
+            if (isDirectory(TAR_INSTALLATION_DIR)) {
+                throw new PrestoManagerException(format("Directory '%s' already exist", TAR_INSTALLATION_DIR.toString()));
+            }
+            else {
+                createDirectories(TAR_INSTALLATION_DIR);
+            }
+            untarFile(tarFile, TAR_INSTALLATION_DIR);
+            createDirectories(CONFIG_DIR);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to install Presto");
+        }
+    }
+
+    public void uninstallAsync(boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        try {
+            deleteRecursively(TAR_INSTALLATION_DIR);
+            deleteRecursively(TAR_DATA_DIR);
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to uninstall Presto", e);
+        }
+        LOGGER.debug("Successfully uninstalled Presto");
+    }
+
+    public void upgradePackage(File tarFile, boolean checkDependencies)
+            throws PrestoManagerException
+    {
+        uninstallAsync(checkDependencies);
+        tarInstall(tarFile);
+    }
+
+    public File getPackageFile(URL packageUrl)
+            throws PrestoManagerException
+    {
+        File tarTempFile;
+        try {
+            LOGGER.debug("Downloading file from url: %s", packageUrl);
+            tarTempFile = downloadFile(packageUrl).toFile();
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException(format("Failed to download the file from url : %s", packageUrl.toString()));
+        }
+        return tarTempFile;
+    }
+
+    public void startAsync()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
+    }
+
+    public void terminate()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
+    }
+
+    public void kill()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
+    }
+
+    public void restartAsync()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
+    }
+
+    public String getVersion()
+            throws PrestoManagerException
+    {
+        throw new PrestoManagerException("Unsupported operation");
```

Comment:
If any of these next few functions are not supported by tarball installations, I recommend that the return type be changed to `Optional<current return type>`. For tarball installation, you can return `Optional.empty()`.

Then, when you add them to the JSON, do it like this:

```java
Optional<String> version = getVersion();
version.ifPresent(version -> json.addProperty("version", version));
```
However, I would hold off on this change until after the current problems with JSON parsing are resolved.